packages/opentelemetry-resources/src/detectors/BrowserDetector.ts

```diff
@@ -14,17 +14,57 @@
  * limitations under the License.
  */
 
-import { IResource } from '../IResource';
+import { Attributes, diag } from '@opentelemetry/api';
+import {
+  SEMRESATTRS_PROCESS_RUNTIME_DESCRIPTION,
+  SEMRESATTRS_PROCESS_RUNTIME_NAME,
+  SEMRESATTRS_PROCESS_RUNTIME_VERSION,
+} from '@opentelemetry/semantic-conventions';
 import { ResourceDetectionConfig } from '../config';
-import { Detector } from '../types';
-import { browserDetectorSync } from './BrowserDetectorSync';
+import { ResourceDetector } from '../types';
 
 /**
  * BrowserDetector will be used to detect the resources related to browser.
  */
-class BrowserDetector implements Detector {
-  detect(config?: ResourceDetectionConfig): Promise<IResource> {
-    return Promise.resolve(browserDetectorSync.detect(config));
+class BrowserDetector implements ResourceDetector {
+  detect(config?: ResourceDetectionConfig) {
+    const isBrowser =
+      typeof navigator !== 'undefined' &&
+      global.process?.versions?.node === undefined && // Node.js v21 adds `navigator`
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore don't have Bun types
```

Comment:
Would `// @ts-expect-error` work here? That would avoid needing to disable the eslint rule right?